pkg/apis/cr/v1alpha1/types.go

```diff
@@ -225,77 +254,112 @@ type BlueprintPhase struct {
 type BlueprintList struct {
 	metav1.TypeMeta `json:",inline"`
 	metav1.ListMeta `json:"metadata"`
-	Items           []*Blueprint `json:"items"`
+	// Items is the list of Blueprints.
+	Items []*Blueprint `json:"items"`
 }
 
 // +genclient
 // +genclient:noStatus
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// Profile
+// Profile captures information about a location for data operation artifacts and
+// corresponding credentials that will be made available to a Blueprint.
 type Profile struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata"`
-	Location          Location   `json:"location"`
-	Credential        Credential `json:"credential"`
-	SkipSSLVerify     bool       `json:"skipSSLVerify"`
+	// Specifies the location that the Blueprint can use.
+	Location Location `json:"location"`
+	// Represents the credentials associated with the Location.
+	Credential Credential `json:"credential"`
+	// SkipSSLVerify is boolean and specifies whether skipping SkipSSLVerify verification
+	// is allowed when operating with the Location.
+	// If omitted from a CR definition it default to false
+	SkipSSLVerify bool `json:"skipSSLVerify"`
 }
 
 // LocationType
 type LocationType string
 
 const (
-	LocationTypeGCS         LocationType = "gcs"
+	// LocationType is selected as GCS.
+	LocationTypeGCS LocationType = "gcs"
+	// LocationType is selected as S3 Complaint.
 	LocationTypeS3Compliant LocationType = "s3Compliant"
-	LocationTypeAzure       LocationType = "azure"
-	LocationTypeKopia       LocationType = "kopia"
+	// LocationType is selected as Azure.
+	LocationTypeAzure LocationType = "azure"
+	// LocationType is selected as Kopia.
+	LocationTypeKopia LocationType = "kopia"
```

Comment:
do we have to add comments here as well. If it's not mandatory let's not add comment to  these. Because of the same reason that it's very self explanatory.

Reply:
Alright, removing comments from here.